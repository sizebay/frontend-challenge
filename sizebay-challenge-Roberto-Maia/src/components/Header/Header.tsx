import moment from "moment";
import {
  DayOfWeek,
  DivDate,
  DivHeader,
  MonthAndYear,
  PDay,
  PMonth,
  PYear,
} from "./styles";

export const Header = () => {
  return (
    <DivHeader>
      <DivDate>
        <PDay>{moment().date()}</PDay>
        <MonthAndYear>
          <PMonth>{moment().format("MMM")}</PMonth>
          <PYear>{moment().format("YYYY")}</PYear>
        </MonthAndYear>
      </DivDate>
      <DayOfWeek>{moment().format("dddd")}</DayOfWeek>
    </DivHeader>
  );
};
