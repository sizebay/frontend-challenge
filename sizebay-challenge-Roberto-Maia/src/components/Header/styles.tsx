import styled from "styled-components";

export const DivHeader = styled.div`
  width: 680px;
  height: 79px;
  opacity: 1;
  display: flex;
  align-items: center;
  justify-content: space-between;

  @media (max-width: 600px) {
    width: 100%;
    height: 79px;
  }
`;

export const DayOfWeek = styled.div`
  width: 124px;
  height: 32px;
  text-align: center;
  font: normal normal normal 24px/32px Roboto;
  letter-spacing: 0px;
  color: #848484;
  opacity: 1;
`;

export const DivDate = styled.div`
  width: 121px;
  height: 79px;
  opacity: 1;
  display: flex;
  gap: 5px;
`;

export const PDay = styled.div`
  width: 208px;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 79px;
  font-size: 70px;
  font-family: Roboto;
  letter-spacing: 0px;
  color: #848484;
  opacity: 1;
`;

export const MonthAndYear = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  letter-spacing: 0px;
  color: #848484;
  opacity: 1;
`;

export const PMonth = styled.div`
  width: 32px;
  height: fit-content;
  padding: 0;
  font-size: 20px;
  letter-spacing: 0px;
  color: #848484;
  opacity: 1;
`;

export const PYear = styled.div`
  width: 53px;
  height: fit-content;
  font-size: 25px;
  letter-spacing: 0px;
  padding: 0;
  color: #848484;
  opacity: 1;
`;
