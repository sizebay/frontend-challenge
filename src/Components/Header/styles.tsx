import styled from 'styled-components';
import { theme } from '../../GlobaStyles/theme';

export const HeaderContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  margin-bottom: 24px;
`;

export const DateHolder = styled.div`
  display: grid;
  justify-content: center;
  grid-template-areas:
    "DateNumber DateMonth"
    "DateNumber DateYear";
`;

export const DateNumber = styled.p`
  grid-area: DateNumber;
  font-size: ${theme.font.size[60]};
  font-weight: ${theme.font.weight.medium};
`;

export const DateMonth = styled.span`
  grid-area: DateMonth;
  align-self: end;
  font-size: ${theme.font.size[24]};
  font-weight: ${theme.font.weight.regular};
  width: 3ch;
  white-space: nowrap;
  overflow: hidden;
`;

export const DateYear = styled.span`
  grid-area: DateYear;
  font-size: ${theme.font.size[24]};
  font-weight: ${theme.font.weight.light};
`;

export const WeekDay = styled.p`
  font-size: ${theme.font.size[24]};
  font-weight: ${theme.font.weight.regular};
`;
