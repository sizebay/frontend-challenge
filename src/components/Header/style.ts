import styled from 'styled-components';

export const Container = styled.header`
  height: fit-content;
  display: flex;
  font-family: "Roboto";
  justify-content: space-between;
  align-items: flex-end;
  color: #848484;

  .weekday {
    font-size: 1.5rem;
  }
`;

export const DateContainer = styled.div`
  width: fit-content;
  display: grid;
  display: grid;
  grid-template-columns: 1fr 1fr;

  span {
    font-size: 3rem;
    font-weight: 700;
  }

  aside {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: flex-start;

    span {
      font-size: 1.5rem;
      line-height: 1.5rem;
      font-weight: normal;
    }
  }
`;