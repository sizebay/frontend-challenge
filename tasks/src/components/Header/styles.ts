import styled from 'styled-components';

export const Container = styled.header`
  display: flex;
  justify-content: space-between;
  align-items: center;

  width: 100%;
  margin-top: 54px;
`;

export const DateContainer = styled.h3`
  display: flex;
  justify-content: center;
  align-items: center;

  > strong {
    font-size: 60px;
    font-weight: 500;
  }

  > div {
    > strong {
      font-size: 20px;
      font-weight: 400;
    }

    > p {
      font-size: 20px;
      font-weight: 300;
      margin: 0;
    }
  }
`;

export const DayOfWeek = styled.h4`
  font-size: 24px;
  font-weight: 400;
`;
