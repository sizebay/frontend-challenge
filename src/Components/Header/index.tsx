import { getCurrentDate } from '../../Utils/GetCurrentDate';
import { DateHolder, DateMonth, DateNumber, DateYear, HeaderContainer, WeekDay } from './styles';

export function Header() {
    return (
        <HeaderContainer>
            <DateHolder>
                <DateNumber>{String(getCurrentDate().dayNumber).padStart(2, '0')}</DateNumber>
                <DateMonth>{getCurrentDate().monthName}</DateMonth>
                <DateYear>{getCurrentDate().year}</DateYear>
            </DateHolder>
            <WeekDay>{getCurrentDate().dayName}</WeekDay>
        </HeaderContainer>
    );
}