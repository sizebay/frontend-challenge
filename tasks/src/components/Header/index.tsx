import React, { useMemo } from 'react';
import moment from 'moment';

import { Container, DateContainer, DayOfWeek } from './styles';

const Header: React.FC = () => {
  const formattedDate = useMemo(() => {
    return {
      date: {
        day: moment().format('DD'),
        month: moment().format('MMM'),
        year: moment().format('YYYY'),
      },
      dayOfWeek: moment().format('dddd'),
    };
  }, []);

  return (
    <Container>
      <DateContainer>
        <strong>{`${formattedDate.date.day}`}</strong>
        <div>
          <strong>{`${formattedDate.date.month}`}</strong>
          <p>{`${formattedDate.date.year}`}</p>
        </div>
      </DateContainer>
      <DayOfWeek>{formattedDate.dayOfWeek}</DayOfWeek>
    </Container>
  );
};

export default Header;
