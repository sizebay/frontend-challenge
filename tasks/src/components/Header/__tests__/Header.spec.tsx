import { render } from '@testing-library/react';
import moment from 'moment';

import Header from '..';

describe('<Header/>', () => {
  it('should match snapshot', () => {
    const container = render(<Header />);

    expect(container).toMatchSnapshot();
  });

  it('should be able to the current date', () => {
    const { getByText, getByRole } = render(<Header />);

    const currentDate = moment().format('DD MMM YYYY');
    const currentDay = moment().format('dddd');

    const date = getByRole('heading', { name: currentDate });
    const dayOfWeek = getByText(currentDay);

    expect(dayOfWeek).toBeInTheDocument();
    expect(date).toBeInTheDocument();
  });
});
