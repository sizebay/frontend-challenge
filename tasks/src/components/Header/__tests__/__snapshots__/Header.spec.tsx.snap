// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  margin-top: 54px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > strong {
  font-size: 60px;
  font-weight: 500;
}

.c1 > div > strong {
  font-size: 20px;
  font-weight: 400;
}

.c1 > div > p {
  font-size: 20px;
  font-weight: 300;
  margin: 0;
}

.c2 {
  font-size: 24px;
  font-weight: 400;
}

<body>
    <div>
      <header
        class="c0"
      >
        <h3
          class="c1"
        >
          <strong>
            10
          </strong>
          <div>
            <strong>
              Sep
            </strong>
            <p>
              2021
            </p>
          </div>
        </h3>
        <h4
          class="c2"
        >
          Friday
        </h4>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-bdfBQB eOzPlJ"
    >
      <h3
        class="sc-gsTEea UvZWt"
      >
        <strong>
          10
        </strong>
        <div>
          <strong>
            Sep
          </strong>
          <p>
            2021
          </p>
        </div>
      </h3>
      <h4
        class="sc-dlfnuX gfXCJR"
      >
        Friday
      </h4>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
